### ISSUE TITLE
ExpandHeap GC test crashes in GcInfoDecoder::GetStackSlot

### ISSUE THREAD

**Issue by kangaroo** [dotnet/coreclr#512]
Frame & Backtrace:

```
frame #6: 0x000000010155f774 libcoreclr.dylib`GcInfoDecoder::GetStackSlot(this=0x00007fff5fbfba98, spOffset=-64, spBase=GC_FRAMEREG_REL, pRD=0x00007fff5fbfc3b0) + 228 at gcinfodecoder.cpp:1826
   1823	        _ASSERTE( GC_FRAMEREG_REL == spBase );
   1824	        _ASSERTE( NO_STACK_BASE_REGISTER != m_StackBaseRegister );
   1825	
-> 1826	        pObjRef = (OBJECTREF*)((*((SIZE_T*)(GetRegisterSlot( m_StackBaseRegister, pRD )))) + spOffset);
   1827	    }
   1828	
   1829	    return pObjRef;
(lldb) bt
* thread #1: tid = 0x15c6d5, 0x00000001019f7617 libcoreclr.dylib`DBG_DebugBreak + 23 at context.cpp:1349, queue = 'com.apple.main-thread', stop reason = EXC_BREAKPOINT (code=EXC_I386_BPT, subcode=0x0)
    frame #0: 0x00000001019f7617 libcoreclr.dylib`DBG_DebugBreak + 23 at context.cpp:1349
    frame #1: 0x0000000101a25da3 libcoreclr.dylib`DebugBreak + 355 at debug.cpp:390
    frame #2: 0x000000010104f63a libcoreclr.dylib`DbgAssertDialog(szFile=0x0000000101c16bc0, iLine=4628, szExpr=0x0000000101c18a01) + 330 at debug.cpp:802
    frame #3: 0x0000000101535be5 libcoreclr.dylib`HandleHardwareException(ex=0x00007fff5fbfaad0) + 373 at exceptionhandling.cpp:4628
    frame #4: 0x0000000101b12cbb libcoreclr.dylib`PAL_DispatchException(dwRDI=140734799788696, dwRSI=5, dwRDX=140734799791024, dwRCX=5, dwR8=0, dwR9=0, pContext=0x00007fff5fbfb160, pExRecord=0x00007fff5fbfb0c8) + 203 at machexception.cpp:659
    frame #5: 0x0000000101b11362 libcoreclr.dylib`PAL_DispatchExceptionWrapper + 6 at dispatchexceptionwrapper.S:33
  * frame #6: 0x000000010155f774 libcoreclr.dylib`GcInfoDecoder::GetStackSlot(this=0x00007fff5fbfba98, spOffset=-64, spBase=GC_FRAMEREG_REL, pRD=0x00007fff5fbfc3b0) + 228 at gcinfodecoder.cpp:1826
    frame #7: 0x000000010155f627 libcoreclr.dylib`GcInfoDecoder::IsScratchStackSlot(this=0x00007fff5fbfba98, spOffset=-64, spBase=GC_FRAMEREG_REL, pRD=0x00007fff5fbfc3b0) + 103 at gcinfodecoder.cpp:1498
    frame #8: 0x00000001015604a7 libcoreclr.dylib`GcInfoDecoder::ReportSlotToGC(this=0x00007fff5fbfba98, slotDecoder=0x00007fff5fbfb8d0, slotIndex=3, pRD=0x00007fff5fbfc3b0, reportScratchSlots=false, inputFlags=0, pCallBack=0x0000000101558040, hCallBack=0x00007fff5fbfd4d0)(void*, OBJECTREF*, unsigned int), void*) + 359 at gcinfodecoder.h:639
    frame #9: 0x000000010155e48c libcoreclr.dylib`GcInfoDecoder::EnumerateLiveSlots(this=0x00007fff5fbfba98, pRD=0x00007fff5fbfc3b0, reportScratchSlots=false, inputFlags=0, pCallBack=0x0000000101558040, hCallBack=0x00007fff5fbfd4d0)(void*, OBJECTREF*, unsigned int), void*) + 4124 at gcinfodecoder.cpp:1019
    frame #10: 0x00000001011ef5ba libcoreclr.dylib`EECodeManager::EnumGcRefs(this=0x0000000100308ef8, pRD=0x00007fff5fbfc3b0, pCodeInfo=0x00007fff5fbfc208, flags=0, pCallBack=0x0000000101558040, hCallBack=0x00007fff5fbfd4d0)(void*, OBJECTREF*, unsigned int), void*) + 842 at eetwain.cpp:4859
    frame #11: 0x00000001015585c7 libcoreclr.dylib`GcStackCrawlCallBack(pCF=0x00007fff5fbfbfd0, pData=0x00007fff5fbfd4d0) + 999 at gcenv.cpp:216
    frame #12: 0x000000010141bef7 libcoreclr.dylib`Thread::MakeStackwalkerCallback(this=0x000000010300ba18, pCF=0x00007fff5fbfbfd0, pCallback=0x00000001015581e0, pData=0x00007fff5fbfd4d0, uFramesProcessed=2)(CrawlFrame*, void*), void*, unsigned int) + 119 at stackwalk.cpp:893
    frame #13: 0x000000010141c13e libcoreclr.dylib`Thread::StackWalkFramesEx(this=0x000000010300ba18, pRD=0x00007fff5fbfc3b0, pCallback=0x00000001015581e0, pData=0x00007fff5fbfd4d0, flags=34048, pStartFrame=0x0000000000000000)(CrawlFrame*, void*), void*, unsigned int, Frame*) + 462 at stackwalk.cpp:974
    frame #14: 0x000000010141cb51 libcoreclr.dylib`Thread::StackWalkFrames(this=0x000000010300ba18, pCallback=0x00000001015581e0, pData=0x00007fff5fbfd4d0, flags=34048, pStartFrame=0x0000000000000000)(CrawlFrame*, void*), void*, unsigned int, Frame*) + 897 at stackwalk.cpp:1057
    frame #15: 0x000000010155908a libcoreclr.dylib`GCToEEInterface::ScanStackRoots(pThread=0x000000010300ba18, fn=0x00000001013133f0, sc=0x00007fff5fbfd6d8)(Object**, ScanContext*, unsigned int), ScanContext*) + 538 at gcenv.cpp:467
    frame #16: 0x000000010121f12b libcoreclr.dylib`CNameSpace::GcScanRoots(fn=0x00000001013133f0, condemned=0, max_gen=2, sc=0x00007fff5fbfd6d8)(Object**, ScanContext*, unsigned int), int, int, ScanContext*) + 507 at gcscan.cpp:191
    frame #17: 0x00000001012f7394 libcoreclr.dylib`WKS::gc_heap::mark_phase(condemned_gen_number=0, mark_only_p=0) + 1604 at gc.cpp:18929
    frame #18: 0x00000001012f0831 libcoreclr.dylib`WKS::gc_heap::gc1() + 481 at gc.cpp:14719
    frame #19: 0x0000000101307cec libcoreclr.dylib`WKS::gc_heap::garbage_collect(n=0) + 3052 at gc.cpp:16236
    frame #20: 0x00000001012e6f1f libcoreclr.dylib`WKS::GCHeap::GarbageCollectGeneration(this=0x000000010020bc68, gen=0, reason=reason_alloc_soh) + 1455 at gc.cpp:34485
    frame #21: 0x00000001012ea0dc libcoreclr.dylib`WKS::gc_heap::try_allocate_more_space(acontext=0x000000010300ba78, size=40, gen_number=0) + 1116 at gc.cpp:12554
    frame #22: 0x00000001012ea203 libcoreclr.dylib`WKS::gc_heap::allocate_more_space(acontext=0x000000010300ba78, size=40, alloc_generation_number=0) + 35 at gc.cpp:12862
    frame #23: 0x0000000101356f50 libcoreclr.dylib`WKS::gc_heap::allocate(jsize=34, acontext=0x000000010300ba78) + 272 at gc.cpp:12893
    frame #24: 0x0000000101343f9a libcoreclr.dylib`WKS::GCHeap::Alloc(this=0x000000010020bc68, acontext=0x000000010300ba78, size=34, flags=0) + 170 at gc.cpp:33955
    frame #25: 0x000000010155bcd5 libcoreclr.dylib`Alloc(size=34, bFinalize=0, bContainsPointers=0) + 277 at gchelpers.cpp:106
    frame #26: 0x000000010155aadc libcoreclr.dylib`FastAllocatePrimitiveArray(pMT=0x000000011c27e8b0, cElements=10, bAllocateInLargeHeap=0) + 668 at gchelpers.cpp:642
    frame #27: 0x000000010158ff9b libcoreclr.dylib`JIT_NewArr1(arrayTypeHnd_=0x000000011c291472, size=10) + 1931 at jithelpers.cpp:3056
    frame #28: 0x000000011c62667a
    frame #29: 0x000000011c6238da
    frame #30: 0x00000001016a79a4 libcoreclr.dylib`CallDescrWorkerInternal + 124 at calldescrworkeramd64.S:900
    frame #31: 0x00000001014c5c22 libcoreclr.dylib`CallDescrWorkerWithHandler(pCallDescrData=0x00007fff5fbfe380, fCriticalCall=0) + 482 at callhelpers.cpp:88
    frame #32: 0x00000001014c6eba libcoreclr.dylib`MethodDescCallSite::CallTargetWorker(this=0x00007fff5fbfe5f0, pArguments=0x00007fff5fbfe570) + 2426 at callhelpers.cpp:627
    frame #33: 0x0000000101454a9d libcoreclr.dylib`MethodDescCallSite::Call_RetArgSlot(this=0x00007fff5fbfe5f0, pArguments=0x00007fff5fbfe570) + 157 at callhelpers.h:421
    frame #34: 0x00000001011211d0 libcoreclr.dylib`RunMain(this=0x00007fff5fbfe708, pParam=0x00007fff5fbfe7a0)::$_0::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*) const::Param*) const::'lambda'(RunMain(MethodDesc*, short, int*, REF<PtrArray>*) const::Param*)::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*) const::Param*) const + 720 at assembly.cpp:2544
    frame #35: 0x000000010111e46c libcoreclr.dylib`RunMain(this=0x00007fff5fbfe790, __EXparam=0x00007fff5fbfe7a0)::$_0::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*) const + 76 at assembly.cpp:2558
    frame #36: 0x000000010111e303 libcoreclr.dylib`RunMain(pFD=0x00000001000e55d0, numSkipArgs=1, piRetVal=0x00007fff5fbfe8b8, stringArgs=0x00007fff5fbfec10) + 323 at assembly.cpp:2558
    frame #37: 0x000000010111e638 libcoreclr.dylib`Assembly::ExecuteMainMethod(this=0x0000000100403b18, stringArgs=0x00007fff5fbfec10) + 328 at assembly.cpp:2666
    frame #38: 0x00000001011a11b4 libcoreclr.dylib`CorHost2::ExecuteAssembly(this=0x0000000100400728, dwAppDomainId=1, pwzAssemblyPath=0x0000000100314578, argc=0, argv=0x0000000000000000, pReturnValue=0x00007fff5fbff26c) + 948 at corhost.cpp:1294
    frame #39: 0x00000001010049f8 libcoreclr.dylib`ExecuteAssembly(exePath=0x00007fff5fbffc40, coreClrPath=0x0000000100104c70, appDomainFriendlyName=0x0000000100007d7c, propertyCount=4, propertyKeys=0x00007fff5fbff6e0, propertyValues=0x00007fff5fbff6c0, argc=0, argv=0x0000000000000000, managedAssemblyPath=0x0000000100104ba0, entryPointAssemblyName=0x0000000000000000, entryPointTypeName=0x0000000000000000, entryPointMethodName=0x0000000000000000, exitCode=0x00007fff5fbff26c) + 1704 at unixinterface.cpp:195
    frame #40: 0x0000000100003932 corerun`ExecuteManagedAssembly(currentExeAbsolutePath=0x00007fff5fbffc40, clrFilesAbsolutePath=0x0000000100104b50, managedAssemblyAbsolutePath=0x0000000100104ba0, managedAssemblyArgc=0, managedAssemblyArgv=0x0000000000000000) + 3282 at corerun.cpp:338
    frame #41: 0x0000000100004320 corerun`main(argc=2, argv=0x00007fff5fbffb20) + 1968 at corerun.cpp:439
    frame #42: 0x00007fff888e05c9 libdyld.dylib`start + 1

```

**Comment by jkotas** [comment]
I will take a look at this one

**Comment by jkotas** [comment]
Fixed by #527
